src/ui/testrtc-test.html

```diff
@@ -137,8 +138,18 @@
       done: function() {
         this.setProgress(null);
         var success =
-            (this.errorCount === 0 && (this.successCount + this.warningCount) > 0);
-        this.state = (success ? 'success' : (this.isDisabled ? 'disabled' : 'failure'));
+          (this.errorCount + this.warningCount === 0 && this.successCount > 0);
+
+        if (success) {
+          this.state = 'success';
+        } else if (this.errorCount > 0) {
+          this.state = 'failure';
+        } else if (this.warningCount > 0) {
```

Comment:
Remove the failure condition and above and make this one
 "if (warningCount > 0 && errorCount == 0) { warning }"
Then add an else condition to catch everything remaining as error.

E.g. if you had errorCount and warningCount==0, your code at the moment does not sets any state. Which seems bogus to me. So I think using failure as a catch all is a better alternative.